core/store/src/trie/config.rs

```diff
@@ -34,6 +34,8 @@ pub struct TrieConfig {
     /// List of shards we will load into memory.
     pub load_mem_tries_for_shards: Vec<ShardUId>,
     pub load_mem_tries_for_all_shards: bool,
+    /// Whether mem-trie should be loaded for each tracked shard.
+    pub load_mem_tries_for_tracked_shards: bool,
```

Comment:
Do we need two different flags here? Is there a case we would use one but not the other? I fear that having two flags here is going to be confusing.

Reply:
I thought someone might want to force `load_mem_tries_for_all_shards` disregarding which shards it is assigned to.
Also, backward compatibility.
But after your comment I am leaning toward removing `load_mem_tries_for_all_shards`, as no one should ever load memtrie for a shard it does not track.

Reply:
Yeah, let's remove `load_mem_tries_for_all_shards`.